packages/patternfly-4/react-core/src/components/Button/Button.tsx

```diff
@@ -95,8 +98,14 @@ const Button: React.FunctionComponent<ButtonProps & InjectedOuiaProps> = ({
         'data-ouia-component-id': ouiaId || ouiaContext.ouiaId
       })}
     >
-      {icon && variant === ButtonVariant.link && <span className="pf-c-button__icon">{icon}</span>}
-      {children}
+      {icon && variant === ButtonVariant.link && linkPosition === 'left' && (
```

Comment:
Not sure if this is beyond the scope of this PR since this was the previous behavior, but do we need to restrict use of `.pf-c-button__icon` to `ButtonVariant.link`? This is probably a question for @mcarrano or @mceledonia. Seems to me like it could be supported in any button, except maybe the plain variation.

Also should we call it `iconPosition` instead of `linkPosition`?

Reply:
I don't see any reason to restrict it, what do you think @mceledonia ?